src/cpp/_pythia8.cpp

```diff
@@ -237,6 +305,12 @@ PYBIND11_MODULE(_pythia8, m)
                  return result;
              })
 
+        .def("fill_event",
+             [](Pythia &self, py::array_t<int> &pid,
+                              py::array_t<double> &energy)
+             {
+                fill_event(self.event, self.particleData, pid, energy);
+             })     
```

Comment:
You should not have "fill_event" and "fill". "fill_event" is still a bad name, I already argued against it. See my previous point about removing "fill_event" completely. If that's not feasible, then please please choose a more fitting name that does not include the redundant "event" part.

Reply:
`fill_event` was a Pythia method whereas `fill` is Event method. The names were as the were to reflect that fact.  I remove the special case mostly because the functions were the methods for different objects, which didn't look good.

Reply:
I don't understand what you mean. I am indifferent how the function is named in C++, but in Python the name should be short and nice.

Reply:
I meant `pythia.event.fill` is method of `event` and `pythia.fill_event` is method of `pythia`, i.e. similar functionality but related to different objects.